src/components/Login.js

```diff
@@ -1,32 +1,94 @@
+            setMessage("Password can't be empty!");
+            return;
+        }
+        else if (!username && !password) {
+            setMessage("Password and username can't be empty!");
+            return;
+        }
 
-        if (result) {
-            setMessageText('Wrong username or password.');
+        try {
+            const result = await Api.login(username, password);
+
+            if (!result) {
+                setMessage('Wrong username or password.');
+            }
+            else {
+                setMessage(`You've successfully logged in!`);
+                setToken(result);
+            }
+        }
+        catch (err) {
+            setMessage("We couldn't reach our servers, make sure you are connected to internet and try again.");
+        }
+    }
+
+    const togglePasswordVisibility = () => {
+        if (!passwordInput) {
+            return;
+        }
+
+        if (passwordInput.type === "password") {
+            // noinspection JSPrimitiveTypeWrapperUsage
+            passwordInput.type = "text";
+            setShowPassword(true);
+        }
+        else {
+            // noinspection JSPrimitiveTypeWrapperUsage
+            passwordInput.type = "password";
+            setShowPassword(false);
         }
     }
 
     return (
-        <div className="login-container">
-            <div className="login">
-                <h1>{ messageText }</h1>
+        <div className={ useTheme("login-container-container") }>
+            <div className={ useTheme("login-container") }>
+                <h3 className={ useTheme("message") }>{ message }</h3>
 
-                <h4>Username:</h4>
-                <input placeholder="Jozko Mrkvicka" onChange={ e => setUsernameInput(e.target.value) } />
+                <h4 className={ useTheme("input-label") }>Username: </h4>
+                <input className={ useTheme("input") } placeholder="AlbertEinstein69"
+                       onChange={ e => setUsername(e.target.value) } />
 
-                <h4>Password:</h4>
-                <input placeholder="Heslo123" onChange={ e => setPasswordInput(e.target.value) } />
+                <h4 className={ useTheme("input-label") }>Password: </h4>
+                <input className={ useTheme("input") } placeholder="password123" type="password"
+                       onChange={ e => { setPassword(e.target.value); setPasswordInput(e.target); } } />
+                <div className={ useTheme("toggle-password-visibility-container") }>
+                    <div className="toggle-password-visibility-checkbox">
+                        {/* TODO: change colors for both themes */}
+                        <CheckBox
+                            checked={ showPassword }
+                            checkBoxStyle={{
+                                checkedColor: `#34b93d`,
+                                size: 25,
+                                unCheckedColor: `${ darkMode ? "#e0e0e0" : "#939393" }`
```

Comment:
Why isn't `useTheme` used here?

Reply:
Because `useTheme` is ONLY for classNames, not for things like these. Please read the `useTheme` function.

Reply:
Ok